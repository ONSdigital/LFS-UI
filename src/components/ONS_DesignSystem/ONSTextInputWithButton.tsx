import React, {ChangeEvent} from "react";
import "./ONSButton.css";

interface Props {
    // Input Props
    labelID?: string
    inputLabel?: string
    handleInputChange: (e: ChangeEvent<HTMLInputElement>, ...args: any[]) => void
    inputValue: string
    // Button Props
    buttonID?: string
    buttonLabel: string
    onButtonClick: (...props: any[]) => void
    primaryButton: boolean
}

export const ONSTextInputWithButton = (props: Props) => {

    let buttonClassName = "btn btn--small u-flex-align-bottom u-mb-xs " +
        (props.primaryButton ? "" : "btn--secondary ");

    return (
        <div className="grid--flex ">
            <p className="field">
                <label className="label" htmlFor="search-field">{props.inputLabel}</label>
                <input type="text" id={props.labelID}
                       className="input input--text input-type__input u-mr-xs u-mb-xs input--w-10"
                       value={props.inputValue}
                       onChange={(e) => props.handleInputChange(e)}/>
            </p>
            <button type="submit" id={props.buttonID} className={buttonClassName}
                    onClick={props.onButtonClick}>
                <span className="btn__inner">{props.buttonLabel}</span>
            </button>
        </div>
    );
};
