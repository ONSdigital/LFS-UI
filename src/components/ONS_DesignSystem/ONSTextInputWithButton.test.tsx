import React from "react";
import Enzyme, {mount, shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {ONSTextInputWithButton} from "./ONSTextInputWithButton";
import sinon from "sinon";

describe("ONS Text Input With Button Test", () => {
    Enzyme.configure({adapter: new Adapter()});

    const props = {
        // Input Props
        labelID: "label_id",
        inputLabel: "input_label",
        handleInputChange: jest.fn(),
        inputValue: "input_value",
        // Button Props
        buttonID: "btn_id",
        buttonLabel: "btn_label",
        onButtonClick: sinon.spy(),
        primaryButton: true
    };

    const secondaryButtonProps = {
        // Input Props
        labelID: "label_id",
        inputLabel: "input_label",
        handleInputChange: jest.fn(),
        inputValue: "input_value",
        // Button Props
        buttonID: "btn_id",
        buttonLabel: "btn_label",
        onButtonClick: sinon.spy(),
        primaryButton: false
    };

    function wrapper(render: any, props: any) {
        return render(
            <ONSTextInputWithButton labelID={props.labelID}
                                    inputLabel={props.inputLabel}
                                    handleInputChange={props.handleInputChange}
                                    inputValue={props.inputValue}
                                    buttonID={props.buttonID}
                                    buttonLabel={props.buttonLabel}
                                    onButtonClick={props.onButtonClick}
                                    primaryButton={props.primaryButton}/>
        );
    }

    it("matches Snapshot", () => {
        expect(wrapper(shallow, props)).toMatchSnapshot()
    });

    it("should render correctly", () => expect(wrapper(shallow, props).exists()).toEqual(true));

    it("should render with the correct label", () => {
        expect(wrapper(mount, props).find("ONSTextInputWithButton").getElement().props.inputLabel).toEqual(props.inputLabel);
    });

    it("simulates change events", () => {
        //defined
        wrapper(mount, props).find("input").simulate("change");
        expect(props.handleInputChange).toHaveBeenCalled();
    });

    it("simulates click events", () => {
        wrapper(mount, props).find("button.btn").simulate("click");
        expect(props.onButtonClick).toHaveProperty("callCount", 1);
    });

    it("displays secondary button", () => {
        expect(wrapper(mount, secondaryButtonProps).find("button").hasClass("btn--secondary")).toEqual(true);
    });


});