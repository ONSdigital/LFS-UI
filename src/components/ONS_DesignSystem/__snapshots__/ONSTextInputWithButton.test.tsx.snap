// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ONS Text Input With Button Test matches Snapshot 1`] = `ShallowWrapper {}`;
